libs/wingsdk/src/target-tf-aws/app.ts

```diff
@@ -344,4 +358,85 @@ export class App extends CdktfApp {
     this.subnets.private.push(privateSubnet2);
     return this._vpc;
   }
+
+  /**
+   * The ECR Repository for the App
+   */
+  public get ecr(): EcrRepository {
+    if (this._ecr) {
+      return this._ecr;
+    }
+
+    const ecr = new EcrRepository(this, "Ecr", {
+      name: "my-ecr-repo", // TODO: make this configurable
+    });
+
+    this._ecr = ecr;
+    return this._ecr;
+  }
+
+  /**
+   * The ECR Authorization Token for the App
+   */
+  public get ecrAuth(): DataAwsEcrAuthorizationToken {
+    if (this._ecr_auth) {
+      return this._ecr_auth;
+    }
+
+    if (!this._ecr) {
+      this.ecr;
+    }
+
+    const ecrAuth = new DataAwsEcrAuthorizationToken(this, "EcrAuth", {
+      registryId: this.accountId,
+    });
+
+    this._ecr_auth = ecrAuth;
+    return this._ecr_auth;
+  }
+
+  /**
+   * The Docker Provider for the App
+   */
+  public get dockerProvider(): DockerProvider {
+    if (this._dockerProvider) {
+      return this._dockerProvider;
+    }
+
+    if (!this._ecr_auth) {
+      this.ecrAuth;
+    }
+
+    this._dockerProvider = new DockerProvider(this, "DockerProvider", {
+      registryAuth: [
+        {
+          address: this.ecrAuth.proxyEndpoint,
+          username: this.ecrAuth.userName,
+          password: this.ecrAuth.password,
+        }
+      ]
+    });
+
+    return this._dockerProvider;
+  }
+
+  /**
+   * The ECS Cluster for the App
+   */
+  public get ecsCluster(): EcsCluster {
```

Comment:
Kind of feels like this shouldn't be part of the `App`. I foresee different providers for `cloud.Service` (e.g. EKS) and then it would make sense to manage the clusters at the provider level.

So perhaps this singleton should be decoupled from the app:

```js
let cluster = tfaws.EcsCluster.getOrCreate(this);
cluster.haveFun();
```